pkg/logger/logger.go

```diff
@@ -1,68 +1,98 @@
 package logger
 
 import (
+	"github.com/go-logr/logr"
 	"github.com/go-logr/zapr"
 	"github.com/pion/ion-sfu/pkg/buffer"
 	"github.com/pion/ion-sfu/pkg/sfu"
+	"github.com/pion/logging"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
 var (
 	wrappedLogger *zap.SugaredLogger
-	// zap logger
-	defaultLogger *zap.Logger
+
+	overridedLogger = logr.Discard()
+
+	// pion/ion-sfu
+	defaultLogger = logr.Discard()
+	// pion/webrtc, pion/turn
+	defaultFactory logging.LoggerFactory
 )
 
-func GetLogger() *zap.Logger {
-	if defaultLogger == nil {
+func GetLogger() logr.Logger {
+	if defaultLogger == logr.Discard() {
 		InitDevelopment("")
 	}
 	return defaultLogger
 }
 
+func SetLogger(logger logr.Logger) {
+	defaultLogger = logger
+	sfu.Logger = logger.WithName("sfu")
+	buffer.Logger = sfu.Logger
+
+	if l, ok := logger.(logr.CallDepthLogger); ok {
```

Comment:
is there a way to ensure that we could get this depth with LogR, otherwise if the user passes in an incompatible logger, our logs will be attributed to the wrong caller.

Reply:
Can't get depth info from logr.Logger. But default implementation of logr LogSink should have correct depth for calling Info()/Error(), incompatibility is user's coding error.